dlgr/griduniverse/experiment.py

```diff
@@ -1720,8 +1720,117 @@ def analyze(self, data):
         return json.dumps({
             "average_payoff": self.average_payoff(data),
             "average_score": self.average_score(data),
+            "number_of_actions": self.number_of_actions(data),
+            "average_time_to_start": self.average_time_to_start(data),
         })
 
+    def ssplit2(self, seq, splitters):
+        """ Split a list into nested lists on a value (or tuple of values)
+            Found at: https://stackoverflow.com/questions/4322705/split-a-list-into-nested-lists-on-a-value
+            auxilary function to number_of_actions
+        """
+        seq = list(seq)
+        if splitters and seq:
+            splitters = set(splitters).intersection(seq)
+            if splitters:
+                result=[]
+                begin=0
+                for end in range(len(seq)):
+                    if seq[end] in splitters:
+                        if end > begin:
+                            result.append(seq[begin:end])
+                        begin=end+1
+                if begin < len(seq):
+                    result.append(seq[begin:])
+                return result
+        return [seq]
+
+    def number_of_actions_per_round(self, origin_ids, moves):
+        """ Calculate number of moves/player for a specific round
+            auxilary function to number_of_actions
+        """
+        player_move_data = []
+        for player in origin_ids:
+            players_moves = [x for x in moves if x[11] == player] # get all the moves of a player
+            if len(players_moves) != 0:
+                # extract player_id from a move data string
+                # '{"move": "up", "type": "move", "actual": "up", "player_id": 2}'
```

Comment:
Can we use `json.loads` here to avoid string manipulation?

Reply:
Brilliant idea. I have clearly not used json.loads enough times in my life to have thought of this myself. Thanks!